### ISSUE TITLE
Incorrect work of [ReferencePicker] with generic inheritance

### ISSUE THREAD

**Issue by dragon122** [TylerTemp/SaintsField#112]
Same setup as in the ReferencePicker example except we move abstract classes to a parent script:
![IllustrationA](https://github.com/user-attachments/assets/30c323c2-b26c-46e4-b4d8-c4b4da5dced1)

Everything works well, but as soon as we make a generic version out of our classes-holders:

![IllustrationB](https://github.com/user-attachments/assets/d52d2636-80fb-4670-9fcd-f227a374a367)

Whenever the reference picker arrow is clicked, nothing happens and a NullRef exception is thrown pointing to the line 31 in "ReferencePickerAttributeDrawer" (I guess "realType" is null).
Interestingly, if we move only one abstract class to the ParentScript - everything will work as intended.

**Comment by TylerTemp** [comment]
Hi,

Interesting, I believe this is a new feature from Unity that I didn't notice.

What's your Unity version?

When I test on 2022.3.14.f1, it shows nothing (no arrow picker at all) with a different error. The script is as you provided:

```csharp
// ParentScript.cs
namespace SaintsField.Samples.Scripts.IssueAndTesting.Issue.Issue112
{
    public class ParentScript<T>: MonoBehaviour
    {
        [Serializable]
        public abstract class Base1Fruit { }

        [Serializable]
        public abstract class Base2Fruit: Base1Fruit { }
    }
}

// ChildScript.cs
namespace SaintsField.Samples.Scripts.IssueAndTesting.Issue.Issue112
{
    public class ChildScript: ParentScript<float>
    {
        [Serializable]
        public class Apple: Base2Fruit { }

        [Serializable]
        public class Orange: Base2Fruit { }

        [SerializeReference, ReferencePicker] public Base2Fruit item;
    }
}
```

Inspector:

![image](https://github.com/user-attachments/assets/c64f7142-a661-4010-a196-19d8d286d42a)

Error:

![image](https://github.com/user-attachments/assets/00477748-2db1-4d82-8b6e-bea6af923748)

> The field ...ChildScript.item has the [SerializeReference] attribute applied, but is of type ....ParentScript.Base2Fruit\<System.Single\>, which is a generic instance type. This is not supported. You must create a non-generic subclass of your generic instance type and use that as the field type instead.

Removing `ReferencePicker` gives the same error, which means the error came from Unity rather than SaintsField.

So providing your unity version is appreciated.

**Comment by dragon122** [comment]
Tested on Unity 6000.0.25f1 and 6000.1.0a4 (SaintsField 3.9) with the same result.
Thanks

**Comment by TylerTemp** [comment]
Hi

This bug has been fixed in [3.9.1](https://github.com/TylerTemp/SaintsField/releases/tag/3.9.1), please have a try.
